### PR TITLE
Properly dispose the scope.

### PR DESCRIPTION
**Pull request by jackliums**
## Description

The IDocumentClient scope is not properly disposed.

## Related issues
Addresses #558.

## Testing
Ran unit tests and integration tests.

### CODE REVIEW

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs [hunk 1]
```diff
@@ -88,14 +86,17 @@ public async Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobReco
 
             try
             {
-                ResourceResponse<Document> result = await DocumentClient.CreateDocumentAsync(
+                using (IScoped<IDocumentClient> documentClient = _documentClientFactory())
+                {
+                    ResourceResponse<Document> result = await documentClient.Value.CreateDocumentAsync(
                     CollectionUri,
```
Comment (namadabu) [inline src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs]: nit: missing indentation

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs [hunk 2]
```diff
@@ -115,14 +116,17 @@ public async Task<ExportJobOutcome> GetExportJobByIdAsync(string id, Cancellatio
 
             try
             {
-                DocumentResponse<CosmosExportJobRecordWrapper> cosmosExportJobRecord = await DocumentClient.ReadDocumentAsync<CosmosExportJobRecordWrapper>(
+                using (IScoped<IDocumentClient> documentClient = _documentClientFactory())
+                {
+                    DocumentResponse<CosmosExportJobRecordWrapper> cosmosExportJobRecord = await documentClient.Value.ReadDocumentAsync<CosmosExportJobRecordWrapper>(
                     UriFactory.CreateDocumentUri(DatabaseId, CollectionId, id),
```
Comment (namadabu) [inline src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs]: nit: missing indentation

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs [hunk 3]
```diff
@@ -231,9 +238,12 @@ public async Task HardDeleteAsync(ResourceKey key, CancellationToken cancellatio
         {
             EnsureArg.IsNotNull(sqlQuerySpec, nameof(sqlQuerySpec));
 
-            CosmosQueryContext context = new CosmosQueryContext(CollectionUri, sqlQuerySpec, feedOptions);
+            var context = new CosmosQueryContext(CollectionUri, sqlQuerySpec, feedOptions);
 
-            return _cosmosDocumentQueryFactory.Create<T>(DocumentClient, context);
+            using (IScoped<IDocumentClient> documentClient = _documentClientFactory())
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs]: I think `documentClient` should be passed into this method. The parent method should own this scope

Comment (jackliums) [reply src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs]: You are right. I should have caught this. Fixed.

**Pull request merged**